src/systems/systems/src/pneumatic/valve.rs

```diff
@@ -0,0 +1,690 @@
+    }
+}
+
+pub struct PneumaticContainerConnector {
+    fluid_flow: VolumeRate,
+    transfer_speed_factor: Ratio,
+}
+impl PneumaticContainerConnector {
+    const HEAT_CAPACITY_RATIO: f64 = 1.4;
+    const TRANSFER_SPEED: f64 = 5.;
+    const HEAT_TRANSFER_SPEED: f64 = 3.;
+
+    pub fn new() -> Self {
+        Self {
+            fluid_flow: VolumeRate::new::<cubic_meter_per_second>(0.),
+            transfer_speed_factor: Ratio::new::<ratio>(1.),
+        }
+    }
+
+    /// Transfer fluid between two containers based on the pressure difference
+    pub fn update_move_fluid(
+        &mut self,
+        context: &UpdateContext,
+        container_one: &mut impl PneumaticContainer,
+        container_two: &mut impl PneumaticContainer,
+    ) {
+        self.heat_conduction(context, container_one, container_two);
+
+        let equalization_volume: Volume = (container_one
+            .pressure()
+            .get::<pascal>()
+            .powf(1. / Self::HEAT_CAPACITY_RATIO)
+            - container_two
+                .pressure()
+                .get::<pascal>()
+                .powf(1. / Self::HEAT_CAPACITY_RATIO))
+            * container_one.volume()
+            * container_two.volume()
+            / (container_two
+                .pressure()
+                .get::<pascal>()
+                .powf(1. / Self::HEAT_CAPACITY_RATIO)
+                * container_one.volume()
+                + container_one
+                    .pressure()
+                    .get::<pascal>()
+                    .powf(1. / Self::HEAT_CAPACITY_RATIO)
+                    * container_two.volume());
+
+        let fluid_to_move = (self.transfer_speed_factor
+            * equalization_volume
+            * (1. - (-Self::TRANSFER_SPEED * context.delta_as_secs_f64()).exp()))
+        .min(container_one.volume())
+        .max(-container_two.volume());
+
+        self.move_volume(container_one, container_two, fluid_to_move);
+
+        self.fluid_flow = fluid_to_move / context.delta_as_time();
+    }
+
+    /// Transfer heat between two containers depending on the temperature difference
+    fn heat_conduction(
+        &self,
+        context: &UpdateContext,
+        container_one: &mut impl PneumaticContainer,
+        container_two: &mut impl PneumaticContainer,
+    ) {
+        let coefficient = 1e-1;
+
+        let temperature_gradient = TemperatureInterval::new::<temperature_interval::kelvin>(
+            container_one.temperature().get::<kelvin>()
+                - container_two.temperature().get::<kelvin>(),
+        );
+
+        container_one.update_temperature(
+            -coefficient
+                * temperature_gradient
+                * (1. - (-Self::HEAT_TRANSFER_SPEED * context.delta_as_secs_f64()).exp()),
+        );
+        container_two.update_temperature(
+            coefficient
+                * temperature_gradient
+                * (1. - (-Self::HEAT_TRANSFER_SPEED * context.delta_as_secs_f64()).exp()),
+        );
```

Comment:
Note that `x` in my suggestion is not a very good name 😉.
```suggestion
        let x = coefficient
            * temperature_gradient
            * (1. - (-Self::HEAT_TRANSFER_SPEED * context.delta_as_secs_f64()).exp());
        container_one.update_temperature(-x);
        container_two.update_temperature(x);
```